x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/BinaryStringStringProcessor.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.expression.function.scalar.string;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.sql.SqlIllegalArgumentException;
+import org.elasticsearch.xpack.sql.expression.function.scalar.processor.runtime.Processor;
+import org.elasticsearch.xpack.sql.expression.function.scalar.string.BinaryStringStringProcessor.BinaryStringStringOperation;
+
+import java.io.IOException;
+import java.util.function.BiFunction;
+
+public class BinaryStringStringProcessor extends BinaryStringProcessor<BinaryStringStringOperation, String, Number> {
+    
+    public static final String NAME = "ss";
+    
+    public BinaryStringStringProcessor(StreamInput in) throws IOException {
+        super(in, i -> i.readEnum(BinaryStringStringOperation.class));
+    }
+
+    public BinaryStringStringProcessor(Processor left, Processor right, BinaryStringStringOperation operation) {
+        super(left, right, operation);
+    }
+
+    public enum BinaryStringStringOperation implements BiFunction<String, String, Number> {
+        //not implemented yet. Relies on soundex similarity for computation
```

Comment:
Maybe we should throw an exception here rather than return 0? Do we really plan on implementing this? I don't think we should drag the phonetic analysis plugin as a depndency of SQL. I worry that implementing these as scripts is fairly likely to send the wrong message about how to do fancy searching with Elasticsearch-SQL. Namely, you should use Elasticsearch queries for fancy string searches rather than SQL things.

Reply:
You're right @nik9000 . We discussed about this before and the fancy searches should be done with Elasticsearch directly. I'll remove the function from there.